sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs

```diff
@@ -967,10 +993,9 @@ public partial class FilterBlobSegment
         public string ServiceEndpoint { get { throw null; } }
         public string Where { get { throw null; } }
     }
-    public partial class JsonTextConfiguration
+    public partial interface IBlobQueryErrorHandler
```

Comment:
Can this be an event/callback? If not, can this be an abstract class?

Reply:
Yes, we could make it an abstract class.

Reply:
Are you thinking something along the lines of:
```
public abstract class BlobQueryErrorHandler
{
    public void Handle(BlobQueryError error) {}
    public async Task HandleAsync(BlobQueryError error) {}
}
```
Do we need an actual implementation for these methods?

Reply:
why not an event so people dont have to implement abstractions? Something like ProcessErrorAsync in https://apiview.dev/Assemblies/Review/b5d7e3304000449c859c2016f718db52#Azure.Messaging.ServiceBus.ServiceBusProcessor

Reply:
Do we need both a sync and an async Func?  We have to call the error handler for a sync context, so I'm not sure it would be possible to only have an async callback.

https://github.com/Azure/azure-sdk-for-net/pull/11586/files?file-filters%5B%5D=.cs#diff-d4fe976f18bf4daaa5ba8795474f7eefR224

https://github.com/Azure/azure-sdk-for-net/pull/11586/files?file-filters%5B%5D=.cs#diff-d4fe976f18bf4daaa5ba8795474f7eefR158

Reply:
If you are calling it from sync context, make it sync.

Reply:
Done.  I didn't make it an `event`, because I need to be able to refer to the lamda outside of `BlobQueryOptions`.

https://github.com/Azure/azure-sdk-for-net/pull/11586/files#diff-05327b3d0b26fefc045c8c30eda382d1R710

Reply:
Do this 
```
class Foo
{
    internal Action<int> _ev;
    public event Action<int> Ev {
        add {
            _ev += value;
        }
        remove {
            _ev -= value;
        }
    }
}
```

... and then access the field.

Reply:
Switched to Event.